src/controllers/commands/CommandController.cpp

```diff
@@ -786,6 +787,31 @@ void CommandController::initialize(Settings &, Paths &paths)
         }
         return "";
     });
+
+    this->registerCommand("/openurl", [](const QStringList &words,
+                                         const ChannelPtr channel) {
+        if (words.size() < 2)
+        {
+            channel->addMessage(makeSystemMessage("Usage: /openurl <URL>."));
+            return "";
+        }
+
+        QString link(words[1]);
```

Comment:
Some URLs can include unencoded spaces (` `; 0x20) (sometimes can appear when copying from a browser) it will lead to behavior that might be unpredictable: the URL getting chopped off at the first space. I think you might as well take every word from `words[1]` onward joined by spaces. `QUrl::fromUserInput()` (or QUrl *tolerant* parsing for that matter) should automatically encode these spaces for you.

tldr `/openurl duck.com?q=testing test test` opens `http://duck.com?q=testing` instead of `http://duck.com?q=testing%20test%20test`